aspnetcore/mvc/controllers/routing.md

```diff
@@ -3,65 +3,60 @@ title: Routing to controller actions in ASP.NET Core
-
-> [!NOTE]
-> This documentation refers to the default routing system added in ASP.NET Core 3.0 called Endpoint Routing. It is still possible to use controllers with the previous version of routing for compatibility purposes. See the [2.2-3.0 migration guide](xref:migration/22-to-30) for instructions. Refer to the 2.2 version of this document for reference material on the legacy routing system.
-
-## Setting up Routing Middleware
-
-In your *Configure* method you may see code similar to:
-
-```csharp
-app.UseEndpoints(endpoints =>
-{
-   endpoints.MapControllerRoute("default", "{controller=Home}/{action=Index}/{id?}");
-});
-```
+* Are defined in startup code or attributes.
+* Describe how URL paths are matched to [actions](#action).
+* Are used to generate URLs for links. The generated links are typically returned in responses.
 
-Inside the call to `UseEndpoints`, `MapControllerRoute` is used to create a single route, which we'll refer to as the `default` route. Most apps with controllers will use a route with a template similar to the `default` route.
+Actions are either [conventionally-routed](#cr) or [attribute-routed](#ar). Placing a route on the controller or [action](#action) makes it attribute-routed. See [Mixed routing](#routing-mixed-ref-label) for more information.
 
-The route template `"{controller=Home}/{action=Index}/{id?}"` can match a URL path like `/Products/Details/5` and will extract the route values `{ controller = Products, action = Details, id = 5 }` by tokenizing the path. This result in a match if the app has a controller named `ProductsController` and an action `Details`:
+This document:
 
-```csharp
-public class ProductsController : Controller
-{
-   public IActionResult Details(int id) { ... }
-}
-```
+* Explains the interactions between MVC and routing:
+  * How typical MVC apps make use of routing features.
+  * Covers both:
+    * [Conventionally routing](#cr) typically used with controllers and views.
+    * *Attribute routing* used with REST APIs. If you're primarily interested in routing for REST APIs, jump to the [Attribute routing for REST APIs](#ar) section.
+  * See [Routing](xref:fundamentals/routing) for advanced routing details.
+* Refers to the default routing system added in ASP.NET Core 3.0, called endpoint routing. It's possible to use controllers with the previous version of routing for compatibility purposes. See the [2.2-3.0 migration guide](xref:migration/22-to-30) for instructions. Refer to the [2.2 version of this document](https://docs.microsoft.com/aspnet/core/mvc/controllers/routing?view=aspnetcore-2.2) for reference material on the legacy routing system.
 
-Note that in this example, model binding would use the value of `id = 5` to set the `id` parameter to `5` when invoking this action. See the [Model Binding](xref:mvc/models/model-binding) for more details.
+## Set up conventional endpoint routing middleware
```

Comment:
I'd suggest changing this back to `Setting up Routing Middleware`

`conventional endpoint routing middleware` is not a term I want to use

Reply:
Alternative: `Setting up a conventional route`